web-common/src/features/dashboards/url-state/convertURLToExplorePreset.ts

```diff
@@ -64,6 +65,8 @@ export function convertURLToExplorePreset(
     (d) => d.name!,
   );
 
+  searchParams = await decompressUrlParams(searchParams);
```

Comment:
Along the lines of my other comments, I'd propose we pull the conditional out of `decompressUrlParams` and into this function here, e.g. something like:
```ts
export async function convertURLToExploreState(url, ...) {
    let urlState;
    if (url.searchParams.has(ExploreStateURLParams.GzippedParams)) {
        urlState = decompressUrlSearchParams(searchParams)
    } else {
        urlState = url.searchParams;
    }
}
```